src/accelerate/utils/modeling.py

```diff
@@ -642,6 +642,28 @@ def get_max_memory(max_memory: Optional[Dict[Union[int, str], Union[int, str]]]
     for key in max_memory:
         if isinstance(max_memory[key], str):
             max_memory[key] = convert_file_size_to_int(max_memory[key])
+
+    # Need to sort the device by type to make sure that we allocate the gpu first.
+    # As gpu/xpu are represented by int, we need to sort them first.
+    gpu_devices = [k for k in max_memory.keys() if isinstance(k, int)]
+    gpu_devices.sort()
+    # check if gpu/xgpu device are available and if not, throw a warning
+    num_deivces = torch.xpu.device_count() if is_xpu_available() else torch.cuda.device_count()
+    for device in gpu_devices:
+        if device >= num_deivces or device < 0:
+            logger.warning(
+                f"Device {max(gpu_devices)} is not available, available device are {list(range(num_deivces))}"
+            )
```

Comment:
```suggestion
            logger.warning(
                f"Device {device} is not available, available device are {list(range(num_devices))}"
            )
```
This will print several times the same warning instead of printing one per wrong device.